provider/cluster/kube/client.go

```diff
@@ -443,7 +516,8 @@ func (c *client) deploymentsForLease(ctx context.Context, lid mtypes.LeaseID) ([
 		c.log.Error(err.Error())
 		return nil, errors.Wrap(err, ErrInternalError.Error())
 	}
-	if deployments == nil {
+
+	if deployments == nil || 0 == len(deployments.Items) {
```

Comment:
:nail_care: `len(nil) == 0`; you don't need both & it's better to always use the one you added.

I think that the differentiation was intentional - zero deployments is the empty set, not an error; but I don't remember why it was that way.